### ISSUE TITLE
Site Editing: Improve saving flow when dismissing changes

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#27896]
**Describe the bug**

When using the granular options to save various template parts and post content, there's not a way to fully dismiss changes one doesn't want to make in the saving flow.  

**To reproduce**
1. Install Gutenberg 9.6+ and a FSE ready theme.
2. Go to Posts > Add New
3. Make some changes to the post > Publish or Save Draft
4. Use the option to edit the template to switch into template editing mode in the document settings section. 
5. Make some changes to the header, footer, and post content.
6. Select "Apply" and only select the changes to the post content to save. Confirm that decision
7. You'll then be brought back to the post editor but see "Update" instead of "Publish"
8. If you select "Update", it then brings you to a screen to save the changes previously not selected in the various template parts. There's not a way to dismiss these and if you try to navigate away, you're notified that you'll lose changes.  

**Expected behavior**

I expect there to be a way to confirm dismissing changes made in other parts of the template after navigating out of template editing mode.

**Screenshots**

Here's a short video talking through this experience: https://cloudup.com/ckTQ5chiQWM

**Editor version (please complete the following information):**
- WordPress version: 5.6
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Gutenberg
- If the Gutenberg plugin is installed, which version is it? 9.6.2

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser: Chrome
 - Version:87.0.4280.88

**Comment by annezazu** [comment]
This also needs to be considered when it comes to the template part editing flow as well. For example, saving changes as a new template part rather than overwriting the current one.

**Comment by javierarce** [comment]
Maybe I'm missing something, but to me this feels like a bug: after navigating out of the template editing mode, shouldn't the unchecked items be discarded automatically?

**Comment by annezazu** [comment]
I'm not quite sure what the intent is :D Good question though. @jameskoster do you have a sense of this?

**Comment by jameskoster** [comment]
Yeah I think this is an oversight. I agree with @javierarce – the unsaved changes should be discarded.

**Comment by annezazu** [comment]
Awesome. Thank you both! I've updated the issue title and description to be more accurate to the bug that remains.

**Comment by annezazu** [comment]
@priethor would you consider this a bug at this point over an enhancement? Wanted to check in first before altering labels :)

**Comment by priethor** [comment]
At this point, it's a fine line dividing both, but I would say it's a bug because I don't believe this behavior was the original intent when designing and developing the feature; it looks like an oversight rather than the original intended purpose (in which case it would be an enhancement because it would indeed be working as expected by those creating the feature).

**Comment by annezazu** [comment]
More usability feedback in general to pass along related to this issue from the [fourteenth call for testing for the FSE Outreach Program:](https://make.wordpress.org/test/2022/04/28/fse-program-testing-call-14-rallying-recipe-reviewers/#comment-2471)

>Still in the template editor I went to view the post. Then trying to get back to the template editor I got the page editor which informed me that there was a saved version that contained changes. (I have been caught by this before – on my own site when making changes in the templates and then going to the page editor, if I had clicked on the revert to saved I potentially might have lost all the changes that I had just tried to make. This is confusing!)

This continues to be a bit of an awkward flow that would benefit from greater clarity.

**Comment by annezazu** [comment]
This was once more mentioned in the [FSE Outreach Program's Site Editor Sneak Peek: ](https://make.wordpress.org/test/2022/11/13/fse-program-exploration-site-editor-sneak-peek/#comment-2765)

>I kept unchecking the box to save the template part, but this preference wasn’t remembered the next time I went to save. I felt like it should remember that (at least until I made another edit to that template part).

**Comment by annezazu** [comment]
Noting that this was brought up in the [FSE Outreach Program's nineteenth call for testing:](https://make.wordpress.org/test/2022/12/14/fse-program-testing-call-19-running-through-refinements/#comment-2846)

>When switching to another template or template part, there’s no message (alert) about unsaved changes will be lost. I know these are saved “in general”, but I think it would be great to prompt save changes before switching to another template or template part.

**Comment by annezazu** [comment]
Noting that this problem also happens with the new "global save" button with browse mode:

https://user-images.githubusercontent.com/26996883/220197326-d24a5f03-864c-49a0-ba0e-cde5cc1edbaa.mov

@jasmussen @youknowriad as a heads up. Notice I can't discard changes and that the push to Save will persist.

**Comment by jasmussen** [comment]
This is a good one to solve. Technically this is mainly surfaced by the new save interface, moreso than being a new issue, insofar as the current way you "discard" changes is you exist the design section without saving. I think this is made especially prominent through the new always visible save button. 

I do agree that discarding changes through an explicit action could be nice to have. Such an option may also, in the future, fit well into a save panel, which I believe @jameskoster and @richtabor have explored some options for, that may be relevant to this conversation. Mainly, once all changes are saved, it seems like the panel could simply say "Saved", just like it does inside the editor.

**Comment by StevenDufresne** [comment]
I've run into this multiple times as well and find this experience to be very frustrating. 

Here's where I got stuck recently.
1. I somehow ended up with multiple navigation menus in my list (I probably did it by clicking around in a template)
2. I clicked on one to delete it.
3. It _randomly_ gave me an Edit Page window
4. I clicked "edit"
5. That somehow made a change that I needed to review
6. I clicked "Review 1 change"
7. Not seeing any revert button,  clicked "cancel", that closed my window
8. I clicked "Review 1 change" again
9. I unchecked "navigation" but now the "Save" button was no longer active.

How do I revert this change?

## GIFs
**Getting a weird edit for doing nothing**
![Screen Capture on 2024-06-04 at 10-46-24 (1)](https://github.com/WordPress/gutenberg/assets/1657336/df208e4e-ef5d-4414-a279-3cc93bc25fa9)

**Not being able to remove it**
![Screen Capture on 2024-06-04 at 10-47-05](https://github.com/WordPress/gutenberg/assets/1657336/45d50826-2799-4294-8eaa-4bdb12341863)

**Environment**
Gutenberg version 18.3.0

**Comment by youknowriad** [comment]
We definitely need a "revert" button on this modal/save flow. Thanks for the feedback.
